2016-02-05 - Doug's best practice

duggiefresh opened the pull request.

michaeldupuisjr commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
````diff
@@ -0,0 +1,85 @@
+How many times have you written something like this?
+
+```js
+
````
Kill the blank lines surrounding the code?

duggiefresh replied:
Sounds good!

michaeldupuisjr commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
````diff
@@ -0,0 +1,85 @@
+```js
+...
+
+farmSentence: computed('animal.noise', 'farmer.name', 'animal.species', 'farmer.location', {
````
I don't think it's necessary to repeat this again.

duggiefresh replied:
I intentionally put this here, because the dependent keys are jumbled. Maybe should place more emphasis on it or kill entirely. What do you think?

michaeldupuisjr replied:
Is your point that the dependent keys are not alphabetical?

linstula replied:
I think the point is that the order of the dependent keys is `animal`, `farmer`, `animal`, `farmer`, rather than `animal`, `animal`, `farmer`, `farmer`. Though, if this is the point, I think it's a bit hard to pick up on.

rsocci commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
```diff
@@ -0,0 +1,85 @@
+const { Component } = Ember;
+
+export default Component.extend({
+  farmSentence: computed('animal.species', 'animal.noise', 'farmer.name', 'farmer.location', {
```
You're destructuring `Component`, but left out `computed`

rsocci commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
```diff
@@ -0,0 +1,85 @@
+const { Component } = Ember;
+
+export default Component.extend({
+  farmSentence: computed('animal.{species,noise}', 'farmer.{name,location}', {
```
You're destructuring `Component`, but left out `computed`

rsocci commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
```diff
@@ -0,0 +1,85 @@
+## Why use Brace Expansion?
+
+Isn't that much nicer? I prefer using brace expansion, because it **organizes** the dependent keys and
+makes it **easier to read**. Goodness forbid a coworker of your's writes dependent keys like this:
```
no apostrophe in `yours`

rsocci commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
```diff
@@ -0,0 +1,85 @@
+
+To those new to Ember, hope you learned something new!
+And to those experienced in Ember, hope this was a refresher!
+And to my coworkers that don't use brace expansions, shame on you!
```
lol

brzpegasus commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
```diff
@@ -0,0 +1,85 @@
+export default Component.extend({
+  farmSentence: computed('animal.species', 'animal.noise', 'farmer.name', 'farmer.location', {
+    get() {
+      return `At ${farmer.location}, Farmer ${farmer.name} owns a ${animal.species} that says "${animal.noise}!"`;
```
`farmer` and `animal` are undefined here. You need to `get` them from `this` first!

brzpegasus commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
```diff
@@ -0,0 +1,85 @@
+export default Component.extend({
+  farmSentence: computed('animal.{species,noise}', 'farmer.{name,location}', {
+    get() {
+      return `At ${farmer.location}, Farmer ${farmer.name} owns a ${animal.species} that says "${animal.noise}!"`;
```
Same as above. You need to do `get(this, 'farmer.location')`, not just directly reference `farmer.location`.

duggiefresh commented:
Woop woop. Rebased and addressed comments... all before the Superb owl. Thanks all

![](http://www.footballz.org/wp-content/uploads/2013/02/superbowl.jpg)

brzpegasus commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
```diff
@@ -0,0 +1,97 @@
+export default Component.extend({
+  farmSentence: computed('animal.species', 'animal.noise', 'farmer.name', 'farmer.location', {
+    get() {
+      let animal = get(this, 'animal');
```
You now need to import `get`.

brzpegasus commented on source/posts/2016-02-05-ember-practices-property-brace-expansion.md:
```diff
@@ -0,0 +1,97 @@
+
+const {
+  Component,
+  computed
```
Missing `get`.
